File: all-submissions/0070-climbing-stairs/solution.java
class Solution {

    public int climbStairsRecur(int n, int i, int[]dp) {
        if(i == n) return 1;
        if(i== n-1) return 1;
        if(dp[i] != 0) return dp[i];
        dp[i] =  climbStairsRecur(n, i+1, dp)  + climbStairsRecur(n, i+2, dp);
        return dp[i];

    }

    public int climbStairs(int n) {
        int[] dp = new int[n];
        return climbStairsRecur(n, 0, dp);

    }
}

File: all-submissions/0070-climbing-stairs/solution.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        dp = {}

        def dfs(currStep):
            if currStep > n:
                return 0
            if currStep == n:
                return 1
            if currStep in dp:
                return dp[currStep]
            dp[currStep] = dfs(currStep+1) + dfs(currStep+2)
            return dp[currStep]
        
        return dfs(0)

